### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by kderme**
Main changes:
- localStateQuery improvements.
- ledger-specs using type families (Core.TxOut instead of Shelley.TxOut)

### CODE REVIEW

**Review by dcoutts — APPROVED** [review]
LGTM!

File: cardano-db-sync/src/Cardano/DbSync.hs [hunk 1]
```diff
@@ -263,6 +266,14 @@ dbSyncProtocols trce env plugin queryVar ledgerVar _version codecs _connectionId
         (cStateQueryCodec codecs)
         (localStateQueryClientPeer (localStateQueryHandler queryVar))
 
+    versionErrorMsg :: Text
+    versionErrorMsg = Text.concat
+        [ Text.pack $ show version
+        , " is not enough. We need at least "
+        , Text.pack $ show minVersion ]
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync.hs:273]: Closing `]` on the next line under the comma to match the rest of the code.

File: cardano-db-tool/src/Cardano/Db/Tool/Validate/Balance.hs [hunk 2]
```diff
@@ -15,10 +15,10 @@ import           Cardano.Chain.Common (CompactAddress, Lovelace, decodeAddressBa
                    toCompactAddress, unsafeGetLovelace)
 import qualified Cardano.Chain.UTxO as Byron
 
+import qualified Cardano.Ledger.Core as Core
```
Comment (erikd) [inline cardano-db-tool/src/Cardano/Db/Tool/Validate/Balance.hs:18]: The qualified imports in this project are far from optimal and I have been meaning to fix them.

As a start in the right direction, can we import that as `Ledger` ?

File: cardano-db-tool/src/Cardano/Db/Tool/Validate/Balance.hs [hunk 3]
```diff
@@ -63,25 +63,30 @@ getByronBalance addrText utxo = do
         then Just lovelace
         else Nothing
 
-getShelleyBalance
-    :: forall era. ShelleyBased era
-    => Text -> Shelley.UTxO era -> Either Text Word64
+getShelleyBalance :: forall era.
+                     ( Core.TxOut era ~ Shelley.TxOut era -- somewhere in ledger-spec, there is probably a better constraint synonym for these
```
Comment (erikd) [inline cardano-db-tool/src/Cardano/Db/Tool/Validate/Balance.hs:67]: Can you try to match the formatting?

I wish I had a tool that could enforce formatting in a way that is not horrible, I hate brittany and ormul??? or whatever its called.

File: cardano-db-sync/src/Cardano/DbSync.hs [hunk 4]
```diff
@@ -263,6 +266,17 @@ dbSyncProtocols trce env plugin queryVar ledgerVar _version codecs _connectionId
...
         (localStateQueryClientPeer (localStateQueryHandler queryVar))
 
+    versionErrorMsg :: Text
+    versionErrorMsg = Text.concat
+        [ "The cardano-node version is too old. Please upgrade to a compatible "
+        , "cardano-node version. The db-sync requires a node that supports "
+        , textShow minVersion
+        , ", the one in use only supports "
+        , textShow version
+        ]
```
Comment (dcoutts) [inline cardano-db-sync/src/Cardano/DbSync.hs:276]: :+1:

**Pull request merged**